File: crates/cli/src/utils/colors.rs
use core_dtos::prelude::*;
use tui::style::Color;

pub fn to_terminal_color(next: &CellColors) -> Color {
    match next {
        CellColors::Rgb(r, g, b) => Color::Rgb(*r, *g, *b),
    }
}

File: crates/app/src/utils/colors.ts
import { CellColors } from "../shared";

export const numberToHex = (value: number) => {
  return value.toString(16);
}

export const rgbToHex = (color: CellColors): string => {
  if (color == null || color.rgb == null) {
    return "#000000";
  }

  const next = (color.rgb || []).reduce((result, v) => {
    return result + numberToHex(v);
  }, "#");

  return next;
}